src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -496,12 +496,23 @@ public override void RegisterOnTermination(Action code)
         /// A <see cref="Task"/> that will complete once the actor system has finished terminating and all actors are stopped.
         /// </returns>
         public override Task Terminate()
+        {
+            if(Settings.CoordinatedShutdownRunByActorSystemTerminate)
+            {
+                CoordinatedShutdown.Get(this).Run(CoordinatedShutdown.ActorSystemTerminateReason.Instance);
+            } else
+            {
+                FinalTerminate();
+            }
+            return WhenTerminated;
+        }
+
+        internal override void FinalTerminate()
```

Comment:
Makes sense

Reply:
This line breaks all UnitTests where we Mock ActorSystem using NSubstitute. Do you think it could be feasible to include `[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]` into AssemblyInfo.cs of Akka project? that solves the issue. Thanks!